listenbrainz_spark/stats/user/listening_activity.py

```diff
@@ -60,6 +80,12 @@ def get_time_range(stats_range: str) -> Tuple[datetime, datetime, relativedelta,
         # compute listening activity for each day but no weekday
         step = relativedelta(days=+1)
         date_format = "%d %B %Y"
+    elif stats_range == "quarter":
+        from_offset = _get_quarter_offset(latest_listen_date)
+        to_offset = relativedelta(months=+6)
```

Comment:
I see that all ranges in this method seem to compute "2 ranges back", but by reading the code and docs I don't understand why this is needed. Is there documetnation or a comment explaining why this is done and what behaviour it has?

Reply:
Explained in the comment above. I'll update documentation of code as well.

Reply:
Added docs to get_listening_activity.  Let me know if I should add more details. https://github.com/metabrainz/listenbrainz-server/pull/1629/files#diff-749661f792d900d163f3d9a7cabd382211ec8b2f886a525e018f5741f83cceb6R161-R165